### PR TITLE
Fix backpropagation order in optimization tutorial

### LINKED ISSUE #3507: Feedback about Optimizing Model Parameters Page
There is the following issue on this page: https://docs.pytorch.org/tutorials/beginner/basics/optimization_tutorial.html

Within the section [Full implementation](https://docs.pytorch.org/tutorials/beginner/basics/optimization_tutorial.html#full-implementation), the loop does not contain the `zero_grad` function on top of the backward propagation block as is recommended in the paragraph preceding this section.

Actual code:
```python
# Backpropagation
loss.backward()
optimizer.step()
optimizer.zero_grad()
```
Recommended code:
```python
optimizer.zero_grad()
loss.backward()
optimizer.step()
```

If you could instruct me how to make this change on the documentation, I would be glad to do that.

### PR DESCRIPTION
**Pull request by patrocinio**
Reorder optimizer.zero_grad(), loss.backward(), and optimizer.step() to match the recommended best practice documented in the tutorial.

Fixes #3507

## Checklist

- [ ] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [ ] Only one issue is addressed in this pull request
- [ ] Labels from the issue that this PR is fixing are added to this pull request
- [ ] No unnecessary issues are included into this pull request.

### CODE REVIEW

**Comment by sekyondaMeta** [comment]
@claude
